### PR TITLE
Move J9 specific i386 evaluators to OpenJ9

### LINKED ISSUE #6343: Move OpenJ9 specific i386 evaluators to OpenJ9
There are many evaluators in https://github.com/eclipse/omr/blob/88caea3932581d01a842e55d49029741e7887a04/compiler/x/i386/codegen/OMRTreeEvaluator.cpp
that should be moved to OpenJ9.

### PR DESCRIPTION
**Pull request by BradleyWood**
Signed-off-by: BradleyWood <bradley.wood@ibm.com>

Closes #6343

### CODE REVIEW

**Comment by BradleyWood** [comment]
@dsouzai Would you mind reviewing?

This should have downstream dependency eclipse-openj9/openj9#14925 because of the change at:

https://github.com/BradleyWood/omr/blob/668ee521c2d9a5ee0e6f7e756b9b004bb2779459/compiler/x/i386/codegen/OMRTreeEvaluator.cpp#L5015

File: compiler/x/i386/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -2641,17 +2636,6 @@ TR::Register *OMR::X86::I386::TreeEvaluator::lstoreEvaluator(TR::Node *node, TR:
...
-      TR_OpaqueMethodBlock *caller = node->getOwningMethod();
-      if (isVolatile && caller)
-         {
-         TR_ResolvedMethod *m = comp->fe()->createResolvedMethod(cg->trMemory(), caller, node->getSymbolReference()->getOwningMethod(comp));
-         if (m->getRecognizedMethod() == TR::java_util_concurrent_atomic_AtomicLong_lazySet)
-            {
-            isVolatile = false;
-            }
-         }
-#endif
```
Comment (dsouzai) [inline compiler/x/i386/codegen/OMRTreeEvaluator.cpp:2654]: See https://github.com/eclipse-openj9/openj9/pull/14925#discussion_r853374839

File: compiler/x/i386/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -2849,25 +2833,6 @@ TR::Register *OMR::X86::I386::TreeEvaluator::lstoreEvaluator(TR::Node *node, TR:
...
-            {
-            if (lowMR)
-               lowMR->setIgnoreVolatile();
-            if (highMR)
-               highMR->setIgnoreVolatile();
-            }
-#endif
-         }
-      }
-
```
Comment (dsouzai) [inline compiler/x/i386/codegen/OMRTreeEvaluator.cpp:2870]: See https://github.com/eclipse-openj9/openj9/pull/14925#discussion_r853377242

**Review by dsouzai — CHANGES_REQUESTED** [review]

File: compiler/x/i386/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (BradleyWood) [reply compiler/x/i386/codegen/OMRTreeEvaluator.cpp:2654]: See [force-push](https://github.com/eclipse/omr/compare/668ee521c2d9a5ee0e6f7e756b9b004bb2779459..c6c201c3f909051a1462a5cb20a2060512d94ce1)

**Review by dsouzai — APPROVED** [review]

**Comment by dsouzai** [comment]
jenkins build all

**Comment by dsouzai** [comment]
jenkins build x32linux,win

**Comment by dsouzai** [comment]
Because this is a coordinated merge, gonna do it on Monday so there's no surprises over the weekend.

**Pull request merged**